src/adaptations/device-layer/DeviceControlServer.cpp

```diff
@@ -109,6 +110,21 @@ WEAVE_ERROR DeviceControlServer::OnResetConfig(uint16_t resetFlags)
             ThreadStackMgr().ClearThreadProvision();
 #endif // WEAVE_DEVICE_CONFIG_ENABLE_THREAD
         }
+
+#if WEAVE_DEVICE_CONFIG_ENABLE_JUST_IN_TIME_PROVISIONING
+        // If a credentials config request has been requested, clear the credential
+        // data, if present.
+        if ((resetFlags & kResetConfigFlag_Credentials) != 0)
```

Comment:
I'm thinking this might get people in trouble.  If you clear the op credentials without clearing the service config things are going to get very confused.  Perhaps we should always clear service config if requested to clear op credentials?

Reply:
done